icu4c/source/common/caniter.cpp

```diff
@@ -466,6 +466,9 @@ Hashtable *CanonicalIterator::getEquivalents2(Hashtable *fillinResult, const cha
             Hashtable remainder(status);
             remainder.setValueDeleter(uprv_deleteUObject);
             if (extract(&remainder, cp2, segment, segLen, i, status) == nullptr) {
+                if (U_FAILURE(status)) {
```

Comment:
Wouldn't it be a good idea to also add one of these to the call to `getEquivalents2()` in `getEquivalents()` so that it aborts immediately upon error there too?